### PR TITLE
more converter tests

### PR DESCRIPTION
**Pull request by xeno-by**
/cc @olafurpg @iTakeshi @dveim

### CODE REVIEW

**Comment by xeno-by** [comment]
@all Tomorrow I'm planning to continue porting and fixing tests from https://github.com/liufengyun/eden/blob/master/src/test/scala/dotty/eden/UntpdSuite.scala. Thanks again, @liufengyun!

File: plugin/src/main/scala/org/scalameta/paradise/reflect/LogicalTrees.scala [hunk 1]
```diff
@@ -303,9 +301,11 @@ trait LogicalTrees { self: ReflectToolkit =>
               .set(SelfRole(g.EmptyTree))
           Some(l.Template(Nil, List(lparent), lself, None))
         case g.Block(
-          List(tree @ g.ClassDef(g.Modifiers(FINAL, g.tpnme.EMPTY, Nil), g.TypeName(anon1), Nil, templ)),
-          g.Apply(g.Select(g.New(g.Ident(g.TypeName(anon2))), nme.CONSTRUCTOR), args))
-          if anon1 == tpnme.ANON_CLASS_NAME.toString && anon2 == tpnme.ANON_CLASS_NAME.toString =>
+            List(
+              tree @ g
+                .ClassDef(g.Modifiers(FINAL, g.tpnme.EMPTY, Nil), g.TypeName(anon1), Nil, templ)),
```
Comment (xeno-by) [inline plugin/src/main/scala/org/scalameta/paradise/reflect/LogicalTrees.scala]: This looks weird

File: tests/converter/src/main/scala/ConverterSuite.scala [hunk 2]
```diff
@@ -41,41 +41,47 @@ trait ConverterSuite extends FunSuite {
...
             (x, y) match {
-              case (q"$xlhs $xop [..$xtargs] (..$xargs)", TermApply519(q"$ylhs.$yop", ytargss, Seq(yargs))) =>
-                loop(xlhs, ylhs) && loop(xop, yop) && loop(xtargs, ytargss.flatten) && loop(xargs, yargs)
-              case (q"$xlhs $xop [..$xtargs] (..$xargs)", TermApplyInfixRightAssoc(ylhs, yop, ytargs, yargs)) =>
+              case (q"$xlhs $xop [..$xtargs] (..$xargs)",
+                    TermApply519(q"$ylhs.$yop", ytargss, Seq(yargs))) =>
+                loop(xlhs, ylhs) && loop(xop, yop) && loop(xtargs, ytargss.flatten) && loop(xargs,
+                                                                                            yargs)
+              case (q"$xlhs $xop [..$xtargs] (..$xargs)",
+                    TermApplyInfixRightAssoc(ylhs, yop, ytargs, yargs)) =>
```
Comment (xeno-by) [inline tests/converter/src/main/scala/ConverterSuite.scala]: Not a fan of this formatting. Imo, the pattern match has become harder to read. Wdyt?

Comment (olafurpg) [inline tests/converter/src/main/scala/ConverterSuite.scala]: What is your expected formatting output if it needs to fit the 80 character column limit?

Comment (iTakeshi) [inline tests/converter/src/main/scala/ConverterSuite.scala]: I think so too. `case <patter> =>` should be in a single line, or if it is too long, it should be broken into something like
```scala
case (
  Some(longPattern)
) => 
  expr
```
This seems to clarify the relationship between `case` and `=>`.

**Comment by olafurpg** [comment]
Wow, fantastic progress 👏 Can we add a test case for #59  `trait Service[F[_]]` that was fixed in #63?   

Otherwise LGTM 👍

File: tests/converter/src/main/scala/ConverterSuite.scala [hunk 3]
```diff
@@ -23,7 +24,87 @@ trait ConverterSuite extends FunSuite {
...
+                loop(xstat, ystat)
+              case (q"(..$xargs)", q"$tuple(..$yargs)")
+                  if tuple.syntax.startsWith("scala.Tuple") =>
+                loop(xargs, yargs)
+              case (ctor"$xctor(...${ Seq() })", ctor"$yctor(...${ Seq(Seq()) })") =>
+                loop(xctor, yctor)
+              case (xpat, p"$ypat @ _") =>
+                loop(xpat, ypat)
+              case (p"$xlhs: $xtpe", p"$ylhs @ (_: $ytpe)") =>
+                loop(xlhs, ylhs)
```
Comment (iTakeshi) [inline tests/converter/src/main/scala/ConverterSuite.scala]: This should be `loop(xlhs, ylhs) && loop(xtpe, ytpe)` ?

**Comment by iTakeshi** [comment]
Looks very nice :+1:

**Comment by Dveim** [comment]
Great stuff!

Is anybody going to add more tests on this weekends? If yes, we should distribute cases to avoid double work.

**Comment by xeno-by** [comment]
@Dveim I'll distribute the cases later today.

**Comment by xeno-by** [comment]
If I enable all the tests from https://github.com/liufengyun/eden/blob/master/src/test/scala/dotty/eden/UntpdSuite.scala, then we get 65 test failures in the following categories:
  - [ ] (*) Vals and vars with patterns on the left-hand side
  - [ ] (*) Types
  - [ ] (*) Modifiers
  - [ ] Lambdas
  - [ ] Type definitions
  - [ ] Abstract vals, vars, etc
  - [ ] Imports
  - [ ] Annotations
  - [ ] Context bounds
  - [ ] By-name parameters

  1. Categories are sorted in decreasing order of complexity. 
  1. Those marked with `(*)` will be really complicated, so I'm taking them. 
  1. Please feel free to distribute the rest between yourself. I suggest that someone who wants to work on a given category comments here in this thread and then the category becomes taken.

Wdyt?

**Comment by xeno-by** [comment]
I'm merging this PR to simplify coordination during the weekend.

**Pull request merged**